tests/integration_test.rs

```diff
@@ -0,0 +1,304 @@
+
+        let simple_account = compiled
+            .find_first("SimpleAccount")
+            .expect("could not find contract");
+        let (abi, bytecode, runtime_bytecode) = simple_account
+            .clone()
+            .into_compact_contract()
+            .into_parts_or_default();
+        let simple_account = CompiledContract {
+            abi,
+            bytecode,
+            runtime_bytecode,
+        };
+        Ok(AACompiledContracts {
+            entry_point: entry_point_contract,
+            simple_account: simple_account,
+        })
+    }
+    async fn setup() -> anyhow::Result<TestContest> {
+        let anvil = Anvil::new().spawn();
+        let owner_wallet: LocalWallet = anvil.keys()[0].clone().into();
+        let provider =
+            Provider::<Http>::try_from(anvil.endpoint())?.interval(Duration::from_millis(10u64));
+
+        let client = SignerMiddleware::new(
+            provider,
+            owner_wallet.clone().with_chain_id(anvil.chain_id()),
+        );
+        let client = Arc::new(client);
+        let contracts = _compile_project()?;
+
+        let (entry_point_address, simple_account_address) =
+            contracts.deploy(client.clone()).await?;
+        Ok(TestContest {
+            anvil,
+            entry_point_address,
+            simple_account_address,
+            owner_wallet,
+        })
+    }
+
+    #[tokio::test]
+    async fn setup_local() -> anyhow::Result<()> {
+        let wallet: LocalWallet = MnemonicBuilder::<English>::default()
+            .phrase("test test test test test test test test test test test junk")
+            .build()
+            .unwrap()
+            .into();
+        let provider = Provider::<Http>::try_from("http://127.0.0.1:8545")?
+            .interval(Duration::from_millis(10u64));
+        let chain_id = provider.get_chainid().await?;
+        let client =
+            SignerMiddleware::new(provider, wallet.clone().with_chain_id(chain_id.as_u64()));
+        let client = Arc::new(client);
+        let contracts = _compile_project()?;
+
+        let (entry_point_address, simple_account_address) =
+            contracts.deploy(client.clone()).await?;
+        println!(
+            "en: {:?}, account: {:?}",
+            entry_point_address, simple_account_address
+        );
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn debug_trace() -> anyhow::Result<()> {
+        let provider = Provider::<Http>::try_from("http://127.0.0.1:8545")?
+            .interval(Duration::from_millis(10u64));
+
+        let res = provider
+            .debug_trace_transaction(
+                H256::from_str(
+                    "0x5cf0d42c07b83beb951127914899111759c5d9114aae10fb8768718b65a4e9ff",
+                )
+                .unwrap(),
+                Default::default(),
+            )
+            .await;
+        println!("{:?}", res);
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn interact_with_local() -> anyhow::Result<()> {
```

Comment:
This test does not work for me. I get `thread 'test::interact_with_local' panicked at 'couldnt start anvil: Os { code: 2, kind: NotFound, message: "No such file or directory" }', /home/vid/.cargo/registry/src/github.com-1ecc6299db9ec823/ethers-core-1.0.2/src/utils/anvil.rs:251:37`

Can the call to `setup()` (setting up anvil while it is already set) be the reason?

Reply:
Yes, it's fine now. I tried again and it goes through now, weird. :slightly_smiling_face: